### PR TITLE
STR-24--BE: Keep track of when a new module is available for a user

### PR DESCRIPTION
**Pull request by mbendandour**
# Description
Sends an email at a specific time when a new module available
  
# Testing
Run "php artisan queue:listen --tries=1"
Open another terminal window and Run "redis-server"
Go to postman and send a user_id in the body to the route "/newModuleAvailable"
Look at the terminal window where you ran the queue command. When you hit send on postman, there should be a 5 second delay before you see anything on the terminal

Check to see if an email was sent to you if 'smtp' is in your env files, or check your log if your email is set up to go to log

# Installation
'Are there any dependencies the reviewer needs to download?'

### CODE REVIEW

File: app/Http/Controllers/ModuleProgressController.php [hunk 1]
```diff
@@ -2,10 +2,13 @@
 
 namespace App\Http\Controllers;
 
+use App\Events\ModulesEvent;
+use App\Models\ModuleProgress;
 use Illuminate\Http\Request;
 use App\Contracts\ModuleProgressContract;
-
-
+use App\Jobs\SendNewModuleEmail;
```
Comment (luisjg) [inline app/Http/Controllers/ModuleProgressController.php]: I don't think this file was added to git

**Review by luisjg — DISMISSED** [review]
See comment

File: app/Jobs/SendNewModuleEmail.php [hunk 2]
```diff
@@ -36,7 +36,7 @@ public function handle()
     {
         $mail = $this->user;
 
-        Mail::to((env('RECIEVE_EMAIL')))->send(new GenericEmail($mail));
+        Mail::to((env('RECIEVE_EMAIL')))->send(new NewModuleAvailable($mail));
```
Comment (liadGolan) [inline app/Jobs/SendNewModuleEmail.php]: Should send to student email if on production ( one of the other mail senders has this for students )

**Review by liadGolan — CHANGES_REQUESTED** [review]
The service test should test for failure and the email send should be updated

**Review by liadGolan — APPROVED** [review]
👍

**Review by luisjg — APPROVED** [review]
👍

**Pull request merged**